eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxos.hs

```diff
@@ -437,10 +442,8 @@ lbl2Phase = "2phase"
 -- | Construct a 2-phase predicate check.
 --
 --   Note that 2-phase predicate checks are by definition static.
-(?!##) :: Bool -> PredicateFailure sts -> Rule sts ctx ()
-(?!##) = labeledPred [lblStatic, lbl2Phase]
-
-infix 1 ?!##
+when2Phase :: Rule sts ctx () -> Rule sts ctx ()
+when2Phase = labeled $ lblStatic NE.:| [lbl2Phase]
 
```

Comment:
How about some comment here, explaining exactly what is going on here, and some discussion about the correct use of this function.   One might also explain what NE.:| does.   It looks like a constructor function?

Reply:
@TimSheard It is a type constructor for `NonEmpty`, it comes from `base`: https://downloads.haskell.org/~ghc/8.10.7/docs/html/libraries/base-4.14.3.0/Data-List-NonEmpty.html#t:NonEmpty